make/cmctl.mk

```diff
@@ -0,0 +1,221 @@
+	jq --arg name "$(notdir $<)" \
+		--arg sha256 "$(shell ./hack/ha.sh $<)" \
+		--arg os "linux" \
+		--arg architecture "$*" \
+		'.name = $$name | .sha256 = $$sha256 | .os = $$os | .architecture = $$architecture' \
+		hack/artifact-metadata.template.json > $@
+
+.PHONY: cmctl-darwin
+cmctl-darwin: bin/cmctl/cmctl-darwin-amd64 bin/cmctl/cmctl-darwin-arm64 | bin/cmctl
+
+.PHONY: cmctl-darwin-tarballs
+cmctl-darwin-tarballs: bin/release/cert-manager-cmctl-darwin-amd64.tar.gz bin/release/cert-manager-cmctl-darwin-arm64.tar.gz | bin/release
+
+.PHONY: cmctl-darwin-metadata
+cmctl-darwin-metadata: bin/metadata/cert-manager-cmctl-darwin-amd64.tar.gz.metadata.json bin/metadata/cert-manager-cmctl-darwin-arm64.tar.gz.metadata.json | bin/metadata
+
+bin/cmctl/cmctl-darwin-amd64 bin/cmctl/cmctl-darwin-arm64:  bin/cmctl/cmctl-darwin-%: $(SOURCES) | bin/cmctl
+	GOOS=darwin GOARCH=$* $(GOBUILD) -o $@ $(CMCTL_GOFLAGS) cmd/ctl/main.go
+
+bin/release/cert-manager-cmctl-darwin-amd64.tar.gz bin/release/cert-manager-cmctl-darwin-arm64.tar.gz: bin/release/cert-manager-cmctl-darwin-%.tar.gz:  bin/cmctl/cmctl-darwin-% bin/scratch/cert-manager.license | bin/scratch bin/release
+	$(eval TARDIR := bin/scratch/$(notdir $@))
+	mkdir -p $(TARDIR)
+	cp $< $(TARDIR)/cmctl
+	cp bin/scratch/cert-manager.license $(TARDIR)/LICENSE
+	tar czf $@ -C $(TARDIR) .
+	rm -rf $(TARDIR)
+
+bin/metadata/cert-manager-cmctl-darwin-amd64.tar.gz.metadata.json bin/metadata/cert-manager-cmctl-darwin-arm64.tar.gz.metadata.json: bin/metadata/cert-manager-cmctl-darwin-%.tar.gz.metadata.json: bin/release/cert-manager-cmctl-darwin-%.tar.gz hack/artifact-metadata.template.json | bin/metadata
+	jq --arg name "$(notdir $<)" \
+		--arg sha256 "$(shell ./hack/ha.sh $<)" \
+		--arg os "darwin" \
+		--arg architecture "$*" \
+		'.name = $$name | .sha256 = $$sha256 | .os = $$os | .architecture = $$architecture' \
+		hack/artifact-metadata.template.json > $@
+
+.PHONY: cmctl-windows
+cmctl-windows: bin/cmctl/cmctl-windows-amd64.exe | bin/cmctl
+
+.PHONY: cmctl-windows-tarballs
+cmctl-windows-tarballs: bin/release/cert-manager-cmctl-windows-amd64.tar.gz bin/release/cert-manager-cmctl-windows-amd64.zip | bin/release
+
+.PHONY: cmctl-windows-metadata
+cmctl-windows-metadata: bin/metadata/cert-manager-cmctl-windows-amd64.tar.gz.metadata.json bin/metadata/cert-manager-cmctl-windows-amd64.zip.metadata.json | bin/release
+
+bin/cmctl/cmctl-windows-amd64.exe: $(SOURCES) | bin/cmctl
+	GOOS=windows GOARCH=amd64 $(GOBUILD) -o $@ $(CMCTL_GOFLAGS) cmd/ctl/main.go
+
+bin/release/cert-manager-cmctl-windows-amd64.zip: bin/cmctl/cmctl-windows-amd64.exe bin/scratch/cert-manager.license | bin/scratch bin/release
+	$(eval TARDIR := bin/scratch/$(notdir $@))
+	mkdir -p $(TARDIR)
+	cp $< $(TARDIR)/cmctl.exe
+	cp bin/scratch/cert-manager.license $(TARDIR)/LICENSE
+	pushd $(TARDIR) && zip -r $(notdir $@) . && popd && mv $(TARDIR)/$(notdir $@) $@
```

Comment:
`pushd` is not available on dash. dash is the default `/bin/sh` on Ubuntu. `/bin/sh` is the default shell used by `make`.

Do you think we could instead force the use of bash to avoid this? It is just a matter of adding:

```make
SHELL := /bin/bash
```

Reply:
ah great idea! I actually meant to set the shell to bash and totally forgot.

I had _no idea_ that dash doesn't include pushd, either! Thanks for the learning :grin:

Reply:
dash makes me sad 😢